cogs/challenges.py

```diff
@@ -35,42 +35,69 @@ async def on_raw_reaction_add(self, payload):  # Participant role.
-            
+
             else:  # weekly 2
                 submitted = self.bot.guild.get_role(715676464573317220)
                 submission_channel = self.bot.guild.get_channel(729453201081761862)
 
             if submitted not in message.author.roles:
                 await message.delete()
                 attach = message.attachments and message.attachments[0]
-                
+
                 if not attach:
-                    msg = f"{message.author.mention} make sure to __upload a " \
-                          f"python file__ that only includes the code required " \
-                          f"for the challenge!"
+                    msg = (
+                        f"{message.author.mention} make sure to __upload a "
+                        f"file__ that only includes the code required "
+                        f"for the challenge!"
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
-                
-                content = "```py\n" + (await attach.read()).decode("u8").replace("`", "\u200b`") + "```"
-                if len(content) > 2040:
-                    msg = f"{message.author.mention} attachment can't be __more" \
-                          f" than 2040 characters__."
+
+                filetype = attach.filename.split(".")[-1]
+
+                if len(filetype) > 4 or len(filetype) == 0:  # Most filetypes are 2-3 chars, 4 just to be safe
+                    return await message.channel.send(
+                        f"{message.author.mention} attachement file extension must be between 1 and 4 characters long",
+                        delete_after=10.0,
+                    )
+
+                code = (await attach.read()).decode("u8")
+
+                content = (
+                    f"```{filetype}\n"
+                    + code.replace("`", "\u200b`")
+                    + "```"
+                )
+                if len(content) > 4096:
+                    # 4096 = max embed description size
+
+                    msg = (
+                        f"{message.author.mention} your submission can't be __more"
+                        f" than {4096 - len(filetype) - 7} characters__."
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
 
                 await message.author.add_roles(submitted)
                 embed = discord.Embed(description=content, color=0x36393E)
-                embed.set_author(name=str(message.author), icon_url=message.author.avatar_url)
-                embed.set_footer(text=f'#ID: {message.author.id} • {len(content)-9} chars')
+                embed.set_author(
+                    name=str(message.author), icon_url=message.author.avatar_url
+                )
+                embed.set_footer(
+                    text=f"#ID: {message.author.id} • {len(content)-9} chars • Language: {filetype} • Code Length: {len(code)}"
```

Comment:
Change the old `{len(content) - 9}` to the new length instead of adding at the end

Reply:
frick, didn't notice that